app/controllers/studentLoans/PglAmountController.scala

```diff
@@ -0,0 +1,105 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.studentLoans
+
+import actions.{AuthorisedAction, TaxYearAction}
+import config.{AppConfig, ErrorHandler}
+import forms.{AmountForm, FormUtils}
+import models.User
+import controllers.studentLoans.routes.StudentLoansCYAController
+import models.mongo.EmploymentUserData
+import play.api.data.Form
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import repositories.EmploymentUserDataRepository
+import services.EmploymentSessionService
+import services.studentLoans.StudentLoansService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import utils.{Clock, SessionHelper}
+import views.html.studentLoans.PglAmountView
+
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class PglAmountController @Inject()(implicit val mcc: MessagesControllerComponents,
+                                    authAction: AuthorisedAction,
+                                    val employmentSessionService: EmploymentSessionService,
+                                    val studentLoansService: StudentLoansService,
+                                    employmentUserDataRepository: EmploymentUserDataRepository,
+                                    view: PglAmountView,
+                                    appConfig: AppConfig,
+                                    errorHandler: ErrorHandler,
+                                    ec: ExecutionContext,
+                                    clock: Clock) extends FrontendController(mcc) with I18nSupport with SessionHelper with FormUtils {
+
+  def show(taxYear: Int, employmentId: String): Action[AnyContent] = (authAction andThen TaxYearAction.taxYearAction(taxYear)).async { implicit user =>
+
+    if (appConfig.studentLoansEnabled) {
+      employmentUserDataRepository.find(taxYear, employmentId).map {
+        case Left(_) => errorHandler.internalServerError()
+        case Right(optionCyaData) =>
+          optionCyaData match {
+            case Some(cyaData) =>
+              val pglAmount = cyaData.employment.studentLoans.flatMap(_.pglDeductionAmount)
+              val employerName = cyaData.employment.employmentDetails.employerName
+
+              val form = pglAmount.fold(amountForm(employerName)
+              )(amountForm(employerName).fill _)
+              Ok(view(taxYear, form, employmentId, employerName))
+            case None => Redirect(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
```

Comment:
Think this redirect should be to student loans cya instead as that will create the cya data if there isn't any / handle any other redirects for you.